### PR TITLE
[ADD] documentation for pipelines and steps

### PR DESCRIPTION
**Pull request by ravinkohli**
Addresses #327

### CODE REVIEW

File: autoPyTorch/pipeline/tabular_classification.py [hunk 1]
```diff
@@ -58,21 +75,37 @@ class TabularClassificationPipeline(ClassifierMixin, BasePipeline):
     Args:
         config (Configuration)
             The configuration to evaluate.
-        steps (Optional[List[Tuple[str, autoPyTorchChoice]]]): the list of steps that
-            build the pipeline. If provided, they won't be dynamically produced.
-        include (Optional[Dict[str, Any]]): Allows the caller to specify which configurations
+        steps (Optional[List[Tuple[str, autoPyTorchChoice]]]):
```
Comment (nabenabe0928) [inline autoPyTorch/pipeline/tabular_classification.py:78]: reminder

File: autoPyTorch/pipeline/tabular_classification.py [hunk 2]
```diff
@@ -41,12 +41,29 @@
...
 
 class TabularClassificationPipeline(ClassifierMixin, BasePipeline):
-    """This class is a proof of concept to integrate AutoSklearn Components
+    """
+    This class is a wrapper around `Sklearn Pipeline
+    <https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html>`_
+    to integrate autoPyTorch components and choices for
+    tabular classification tasks.
 
     It implements a pipeline, which includes as steps:
```
Comment (nabenabe0928) [inline autoPyTorch/pipeline/tabular_classification.py:50]:
```suggestion
    It implements a pipeline, which includes the following as steps:
```

File: autoPyTorch/pipeline/components/base_choice.py [hunk 3]
```diff
@@ -205,19 +220,23 @@ def predict(self, X: np.ndarray) -> np.ndarray:
...
 
         Args:
-            X (np.ndarray): input features from which to predict the target
+            X (np.ndarray):
+                input features from which to predict the target
 
         Returns:
-            np.ndarray: the predicted target
+            np.ndarray:
+                the predicted target
```
Comment (nabenabe0928) [inline autoPyTorch/pipeline/components/base_choice.py:228]:
```suggestion
                the target prediction
```

**Review by nabenabe0928 — COMMENTED** [review]
Thanks for the PR.
Your changes look mostly good to me.

File: docs/api.rst [hunk 4]
```diff
@@ -23,4 +23,48 @@ Regression
```
[the remark this answers is not in the document]
Comment (ravinkohli) [reply docs/api.rst:23]: So this file basically contains the API documentation for AutoPyTorch. If you would like to take a look, you can make the docs locally on your laptop and then browse the HTML. 

You need to do 
```
pip install -e .[doc,examples]
cd docs
make html
```

**Pull request merged**